x-pack/functionbeat/magefile.go

```diff
@@ -9,29 +9,59 @@ package main
 import (
 	"context"
 	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
 	"time"
 
 	"github.com/magefile/mage/mg"
-	"github.com/magefile/mage/sh"
 
 	devtools "github.com/elastic/beats/dev-tools/mage"
 	functionbeat "github.com/elastic/beats/x-pack/functionbeat/scripts/mage"
 )
 
+var (
+	availableProviders = []string{
+		"aws",
+	}
+	selectedProviders []string
+)
+
 func init() {
 	devtools.BeatDescription = "Functionbeat is a beat implementation for a serverless architecture."
 	devtools.BeatLicense = "Elastic License"
+	selectedProviders = getConfiguredProviders()
 }
 
 // Build builds the Beat binary.
 func Build() error {
-	return devtools.Build(devtools.DefaultBuildArgs())
+	workingDir, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	params := devtools.DefaultBuildArgs()
+	for _, provider := range selectedProviders {
+		params.Name = devtools.BeatName + "-" + provider
+		err = os.Chdir(workingDir + "/" + provider)
```

Comment:
Changing directories can wreck havoc on the build process (especially tasks running concurrently). I highly recommend trying to add an option to the `BuildArgs` so that you can specify what package to build (e.g. `go build -o functionbeat-aws ./provider/aws`) without needing to change directories.